chromium-x11 / 67.0.3396.62 compile error

Hi,

after recent update chromium-x11 to 67.0.3396.62 fails for me with:

```
FAILED: obj/components/crash/content/app/app/breakpad_linux.o 
arm-angstrom-linux-gnueabi-g++  -march=armv7ve -mthumb -mfpu=neon-vfpv4 -mfloat-abi=hard -mcpu=cortex-a7 --sysroot=/home/superandy/tmp/oe-core-glibc/work/cortexa7t2hf-neon-vfpv4-angstrom-linux-gnueabi/chromium-x11/67.0.3396.62-r0/recipe-sysroot -MMD -MF obj/components/crash/content/app/app/breakpad_linux.o.d -DCRASH_IMPLEMENTATION -DV8_DEPRECATION_WARNINGS -DUSE_UDEV -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_X11=1 -DFULL_SAFE_BROWSING -DSAFE_BROWSING_CSD -DSAFE_BROWSING_DB_LOCAL -DOFFICIAL_BUILD -DCHROMIUM_BUILD -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_32 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_26 -DV8_USE_EXTERNAL_STARTUP_DATA -I../.. -Igen -I../../third_party/crashpad/crashpad -I../../third_party/crashpad/crashpad/compat/linux -I../../third_party/crashpad/crashpad/compat/non_win -I../../third_party/breakpad/breakpad/src -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -funwind-tables -fPIC -pipe -pthread -march=armv7ve -mfloat-abi=hard -mtune=cortex-a7 -mfpu=neon-vfpv4 -mthumb -Wall -Wno-psabi -Wno-unused-local-typedefs -Wno-maybe-uninitialized -Wno-deprecated-declarations -fno-delete-null-pointer-checks -Wno-comments -Wno-missing-field-initializers -Wno-unused-parameter -O2 -fno-ident -fdata-sections -ffunction-sections -fno-omit-frame-pointer -g0 -fvisibility=hidden -isystem../../../recipe-sysroot/usr/include/glib-2.0 -isystem../../../recipe-sysroot/usr/lib/glib-2.0/include -std=gnu++14 -Wno-narrowing -fno-exceptions -fno-rtti -fvisibility-inlines-hidden  -O2 -pipe -feliminate-unused-debug-types -fdebug-prefix-map=/home/superandy/tmp/oe-core-glibc/work/cortexa7t2hf-neon-vfpv4-angstrom-linux-gnueabi/chromium-x11/67.0.3396.62-r0=/usr/src/debug/chromium-x11/67.0.3396.62-r0 -fdebug-prefix-map=/home/superandy/tmp/oe-core-glibc/work/cortexa7t2hf-neon-vfpv4-angstrom-linux-gnueabi/chromium-x11/67.0.3396.62-r0/recipe-sysroot= -fdebug-prefix-map=/home/superandy/tmp/oe-core-glibc/work/cortexa7t2hf-neon-vfpv4-angstrom-linux-gnueabi/chromium-x11/67.0.3396.62-r0/recipe-sysroot-native= -g1 -c ../../components/crash/content/app/breakpad_linux.cc -o obj/components/crash/content/app/app/breakpad_linux.o
In file included from ../../third_party/lss/linux_syscall_support.h:111:0,
                 from ../../third_party/breakpad/breakpad/src/common/memory_allocator.h:50,
                 from ../../third_party/breakpad/breakpad/src/client/linux/dump_writer_common/thread_info.h:37,
                 from ../../third_party/breakpad/breakpad/src/client/linux/minidump_writer/linux_dumper.h:51,
                 from ../../third_party/breakpad/breakpad/src/client/linux/minidump_writer/minidump_writer.h:41,
                 from ../../third_party/breakpad/breakpad/src/client/linux/handler/exception_handler.h:42,
                 from ../../components/crash/content/app/breakpad_linux.cc:48:
../../third_party/crashpad/crashpad/compat/linux/sys/ptrace.h:39:35: error: 'constexpr const __ptrace_request PTRACE_GETVFPREGS' redeclared as different kind of symbol
 static constexpr __ptrace_request PTRACE_GETVFPREGS =
                                   ^~~~~~~~~~~~~~~~~
In file included from /home/superandy/tmp/oe-core-glibc/work/cortexa7t2hf-neon-vfpv4-angstrom-linux-gnueabi/chromium-x11/67.0.3396.62-r0/recipe-sysroot/usr/include/sys/ptrace.h:53:0,
                 from ../../third_party/crashpad/crashpad/compat/linux/sys/ptrace.h:18,
                 from ../../third_party/lss/linux_syscall_support.h:111,
                 from ../../third_party/breakpad/breakpad/src/common/memory_allocator.h:50,
                 from ../../third_party/breakpad/breakpad/src/client/linux/dump_writer_common/thread_info.h:37,
                 from ../../third_party/breakpad/breakpad/src/client/linux/minidump_writer/linux_dumper.h:51,
                 from ../../third_party/breakpad/breakpad/src/client/linux/minidump_writer/minidump_writer.h:41,
                 from ../../third_party/breakpad/breakpad/src/client/linux/handler/exception_handler.h:42,
                 from ../../components/crash/content/app/breakpad_linux.cc:48:
/home/superandy/tmp/oe-core-glibc/work/cortexa7t2hf-neon-vfpv4-angstrom-linux-gnueabi/chromium-x11/67.0.3396.62-r0/recipe-sysroot/usr/include/sys/ptrace-32.h:126:3: note: previous declaration '__ptrace_request PTRACE_GETVFPREGS'
   PTRACE_GETVFPREGS = 27,
   ^~~~~~~~~~~~~~~~~
```

My environment is more or less sumo:
Build Configuration:
BB_VERSION           = "1.37.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "fedora-27"
TARGET_SYS           = "arm-angstrom-linux-gnueabi"
MACHINE              = "raspberrypi3"
DISTRO               = "angstrom"
DISTRO_VERSION       = "v2018.05"
TUNE_FEATURES        = "arm armv7ve vfp thumb neon vfpv4 callconvention-hard cortexa7"
TARGET_FPU           = "hard"
meta-angstrom        = "sumo:0d3fe793b7d486bacd80bf5dacf2a76cf61cbf19"
meta-oe              
meta-filesystems     
meta-gnome           
meta-initramfs       
meta-multimedia      
meta-networking      
meta-python          
meta-webserver       
meta-xfce            = "sumo+work:598198e2dbedbc68b0187eb8b0b246741b5c19d8"
meta-browser         = "master:23b216a2f3202ff7679e5d60d2d7277e8a2b1dcb"
meta-raspi-light     = "master:589053f582d63d6e19bec659e31f3bf3c32b2744"
meta-gumstix-community = "master:91dc39a703fb419e9a131ea4c7c0bf1b74cf5dbc"
meta-bbone           = "master:c0d7810f59a41c6df2b3f00b31d3de551936ab15"
meta-misc            = "master:cdd1a803f98344a50ee5998b598022c5effc15c5"
meta-office          = "master:6fe4ac377d926a747f563c8bd6931674dbe01e7c"
meta-qt5             = "rocko-5.9.4:4e7ba65ba6ad4f88683701428aba037f2091fafc"
meta-qt5-extra       = "master:a341bd5bbda574152fc945e015d40ddc9805acf7"
meta-games           = "work-for-sumo:34e057301589028447cfad83eafe51055379b8a8"
meta-retro           = "master:e6e2d9fbdade54c866fd272b7d232cb14ca5bb81"
meta                 = "sumo+work:94d4078908004dce83d057f57c1fff017788f71a"

OE-Core sumo +
+ 94d4078908004dce83d057f57c1fff017788f71a alsa-lib: do not build with debug output by default

Meta-Oe
+ e329044a67f2215137f521912c6d7e393ce15d03 fluidsynth: update 1.1.10 -> 1.1.11
+ 9b2947190f5dbd74629c907614df64965f2056e6 xfce4-screenshooter: update 1.9.1 -> 1.9.2
+ 67559a7af867f755a547f5fdffbbd7bbc1611534 xfce4-whiskermenu-plugin: update 2.1.6 -> 2.1.7
+ d819a588b9a4a14b7f13d3fb899e970e2a8441fa menulibre: update 2.1.5 -> 2.2.0
+ 3aae63a0da25ec8e798f8939327b4f7858c0f556 xfce4-pulseaudio-plugin: update 0.4.0 -> 0.4.1
+ 820d26867664b3660ba7d6de3ac19076e6af5dd1 parole: update 1.0.0 -> 1.0.1
+ 4290523d137d63063fcead36a36ac0ab481ee2b8 evince: update 3.24.0 -> 3.28.1
+ 598198e2dbedbc68b0187eb8b0b246741b5c19d8 evince: update 3.28.1 -> 3.28.2

--- a/recipes-browser/chromium/chromium-x11_67.0.3396.62.bb
+++ b/recipes-browser/chromium/chromium-x11_67.0.3396.62.bb
@@ -3,6 +3,7 @@ require chromium-gn.inc
 
 SRC_URI += "\
         file://0001-aarch64-Use-xzr-instead-of-x31-in-the-ASM-code.patch \
+        file://0001-Ensure-all-targets-build-when-target_arch-arm-and-target_os-linux.patch \
         file://aarch64-skia-build-fix.patch \
 "
 

--- a/recipes-browser/chromium/files/0001-Ensure-all-targets-build-when-target_arch-arm-and-target_os-linux.patch
+++ b/recipes-browser/chromium/files/0001-Ensure-all-targets-build-when-target_arch-arm-and-target_os-linux.patch
@@ -1,0 +1,37 @@
+From 0ae1244194778385d5fa100c3ebb537b1dfa99f4 Mon Sep 17 00:00:00 2001
+From: Tom Anderson <thomasanderson@chromium.org>
+Date: Wed, 16 May 2018 03:23:04 +0000
+Subject: [PATCH] Ensure all targets build when target_arch="arm" and target_os="linux"
+
+Also requires these third party changes:
+https://chromium-review.googlesource.com/c/crashpad/crashpad/+/1060155
+https://chromium-review.googlesource.com/c/native_client/src/native_client/+/1060158
+
+BUG=843240
+R=thakis
+TBR=wfh
+
+Change-Id: I9288967e238ab5319e1c963fffd58b9fe4be3692
+Reviewed-on: https://chromium-review.googlesource.com/1060129
+Commit-Queue: Thomas Anderson <thomasanderson@chromium.org>
+Reviewed-by: Nico Weber <thakis@chromium.org>
+Cr-Commit-Position: refs/heads/master@{#558953}
+
+Upstream-Status: Pending [1]
+
+[1] https://chromium-review.googlesource.com/c/chromium/src/+/1060129/4/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h
+---
+
+diff --git a/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h b/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h
+index 73861576..e5c95c7 100644
+--- a/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h
++++ b/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h
+@@ -34,7 +34,7 @@
+ #endif  // !PTRACE_GET_THREAD_AREA && !PT_GET_THREAD_AREA && defined(__GLIBC__)
+ 
+ // https://sourceware.org/bugzilla/show_bug.cgi?id=22433
+-#if !defined(PTRACE_GETVFPREGS) && \
++#if !defined(PTRACE_GETVFPREGS) && !defined(PT_GETVFPREGS) && \
+     defined(__GLIBC__) && (defined(__arm__) || defined(__arm64__))
+ static constexpr __ptrace_request PTRACE_GETVFPREGS =
+     static_cast<__ptrace_request>(27);

--- a/recipes-browser/chromium/files/v8-qemu-wrapper.patch
+++ b/recipes-browser/chromium/files/v8-qemu-wrapper.patch
@@ -23,7 +23,7 @@ index dcefe37..e6a8c8f 100644
 --- a/v8/BUILD.gn
 +++ b/v8/BUILD.gn
 @@ -810,6 +810,7 @@ if (v8_use_snapshot) {
-     ]
+     data = []
  
      args = [
 +      "./v8-qemu-wrapper.sh",

